src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxList.WithLotsOfChildren.cs

```diff
@@ -53,7 +53,19 @@ public override int FindSlotIndexContainingOffset(int offset)
             {
                 Debug.Assert(offset >= 0 && offset < FullWidth);
                 int idx = _childOffsets.BinarySearch(offset);
-                return idx >= 0 ? idx : (~idx - 1);
+
```

Comment:
Yeah, I wondered about this case when I wrote it. I originally had a hand-coded binary search that avoided this problem, but I was 'encouraged' to change it to Array.BinarySearch in review.

What we really need is an "upper_bound" equivalent (Array.BinarySearch gives the lower bound).

Note that this BinarySearch is already an extension method over int[]. We could introduce a "UpperBoundBinarySearch" method. That would be nice to avoid code duplication (both C# and VB need the same fix).